File: C++ implementation/linearSearch.cpp
#include <iostream>
#include <time.h>
 
using namespace std;

const int SIZE = 50000;
 
const long int ARRAY_LENGHT = 100000;
const int MAX_RANDOM = 100000;
const int MIN_RANDOM = 1;
 
bool linearSearch(int value, int array[]) {
    bool found = false;
    for(int index=0; index<ARRAY_LENGHT; index++) { 
        cout << " " ;
        if(array[index]==value) {
            found = true;
            break;
        }
    }
    return found;
}
 
int main() {
 
    time_t start, end; 
    time(&start); 
    int array[ARRAY_LENGHT];
 
    for(int index=0; index<ARRAY_LENGHT; index++) {
        array[index] = MIN_RANDOM + rand() % (MAX_RANDOM - MIN_RANDOM + 1);
    }
 
    if(linearSearch(200000, array)) {
        cout << "\nValue was found :)" << endl;
    } else {
        cout << "\nValue was not found :(" << endl;
    }
    
    time(&end); 
    double time_taken = double(end - start); 
    
    const int* memoryI = &SIZE;
    int* memoryF = &array[ARRAY_LENGHT];
    
    cout << "Time taken by program is : " << fixed << time_taken << "sec" << endl;
    cout << "Memory taken by program is : " << memoryF - memoryI;
    
    return 0;
}

File: Go implementation/linearSearch.go
package main

import (
	"fmt"
	"log"
	"math/rand"
	"time"
)

const ARRAY_LENGHT int = 500000
const MAX_RANDOM = 100000
const MIN_RANDOM = 1

func linearSearch(value int, array [ARRAY_LENGHT]int) bool {
	found := false
	for index := 0; index < len(array); index++ {
		fmt.Println(" ")
		if array[index] == value {
			found = true
			break
		}
	}
	return found
}

func main() {
	start := time.Now()
	var array [ARRAY_LENGHT]int

	for index := 0; index < len(array); index++ {
		array[index] = (rand.Intn(MAX_RANDOM-MIN_RANDOM) + MIN_RANDOM)
	}

	if linearSearch(200000, array) {
		fmt.Println("Element was found :)")
	} else {
		fmt.Println("Element wasn´t found :(")
	}

	end := time.Since(start)
	log.Printf("Linear search took %s", end)
}
